terminal.go

```diff
@@ -0,0 +1,49 @@
+//
+// Copyright (c) 2017 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"os"
+
+	"golang.org/x/sys/unix"
```

Comment:
One benefit of importing "github.com/moby/moby/pkg/term" is that we do not have OS specific code.

Reply:
Good point. 

If this is unix specific, we should add build tag 
```
// +build linux
```

or rename this file to `terminal_linux.go`

Reply:
@bergwolf @WeiZhang555 I get the point, but do we care about this code being portable to something else than Linux ?

Reply:
@WeiZhang555 I have renamed the file to `terminal_linux.go`.
@bergwolf I didn't like that we were relying on a docker package for that. Also, this commit actually solves a real issue because we were not checking if STDIN was a terminal, meaning we were failing when trying to set the terminal as RAW in case it was not a terminal.

Reply:
fair enough. Let's take this for now and add other OS support when necessary.